File: 2643. Row With Maximum Ones.js
// 2643. Row With Maximum Ones
mat = [[0,0,0],[0,1,1]]
/**
 * @param {number[][]} mat
 * @return {number[]}
 */
var rowAndMaximumOnes = function(mat) {
  const result = mat.map((row, i) => [i, row.filter(val => val === 1).length]);
  const maxRow = result.reduce((max, curr) => curr[1] > max[1] ? curr : max);
  return [maxRow[0], maxRow[1]];
}
console.log(rowAndMaximumOnes(mat)); // Output: [1, 2]

File: 2643. Row With Maximum Ones.py
#PYTHON
#* 2643. Row With Maximum Ones
mat = [[0,0,0],[0,1,1]]
from typing import List
#* MY SOL | OPT
class Solution:
    def rowAndMaximumOnes(self, mat: List[List[int]]) -> List[int]:
        result = [(i, row.count(1)) for i, row in enumerate(mat)]
        max_row = max(result, key=lambda x: x[1])
        return [max_row[0], max_row[1]]
    
print(Solution().rowAndMaximumOnes(mat))

#* SOL | CONCISE | (1,2)
class Solution:
    def rowAndMaximumOnes(self, mat: List[List[int]]) -> List[int]:
        return max(((i, row.count(1)) for i, row in enumerate(mat)), key=lambda x: x[1])
    
print(Solution().rowAndMaximumOnes(mat))

#* SOL | CONCISE | [1,2]
class Solution:
    def rowAndMaximumOnes(self, mat: List[List[int]]) -> List[int]:
      max_row = max(((i, row.count(1)) for i, row in enumerate(mat)), key=lambda x: x[1])
      return list(max_row)
    
print(Solution().rowAndMaximumOnes(mat))

